controllers/metal3.io/host_state_machine.go

```diff
@@ -428,9 +428,6 @@ func (hsm *hostStateMachine) handleReady(info *reconcileInfo) actionResult {
 }
 
 func (hsm *hostStateMachine) provisioningCancelled() bool {
-	if hsm.Host.Status.ErrorMessage != "" {
```

Comment:
Do we need this condition to remain, for a different path? is there a way for us to check if we are in the Adopt() path?